text_localization_environment/TextLocEnv.py

```diff
@@ -86,8 +86,13 @@ def calculate_reward(self, action):
             else:
                 reward = -self.ETA
         else:
-            new_iou = self.compute_best_iou()
-            reward = np.sign(new_iou - self.iou)
+            new_iou, box = self.compute_best_iou()
+
+            env_box_center = (np.array(self.bbox[2:]) - np.array(self.bbox[0:2]))/2
+            text_box_center = (np.array(self.episode_true_bboxes[box][1]) - np.array(self.episode_true_bboxes[box][0]))/2
```

Comment:
I’m pretty sure that you’re missing to add the top left coordinate of the bounding box to the currently calculated distance of the bottom left coordinate to the box’s center. 

Take, for example, a bounding box defined by (2,2) and (4,4) as its top left and bottom right coordinates. Currently, (to my mind) the center would be calculated as (1,1). However, the correct result would imo be (3,3) – which could be achieved by simply adding the box’s top left coordinate to the current result. 

This applies to the line above as well.